### PR TITLE
Update plugins and dependencies

### PR DESCRIPTION
**Pull request by jaredsburrows**
@ejona86 @zhangkun83 This solves https://github.com/google/protobuf-gradle-plugin/issues/74 and updates many other plugins.

- Updated most plugins
- Updated most dependencies
- Updated some test projects to proto3 syntax

### CODE REVIEW

File: testProject/build.gradle [hunk 1]
```diff
@@ -32,26 +32,21 @@ sourceSets {
   }
 }
 
-def protobufDep = 'com.google.protobuf:protobuf-java:3.0.0-alpha-3'
-def protobufNanoDep = 'com.google.protobuf.nano:protobuf-javanano:3.0.0-alpha-3'
-
 dependencies {
-  compile protobufDep
+  compile 'com.google.protobuf:protobuf-java:3.0.0-alpha-3'
```
Comment (zhangkun83) [inline testProject/build.gradle]: The pre-defined variables makes it easier and less error-prone to upgrade the versions. What's the reason for this change?

File: build.gradle [hunk 2]
```diff
@@ -7,64 +7,57 @@ Base directory: $projectDir
 Running script ${relativePath(buildFile)}
 """
 
-apply plugin: 'idea'
-apply plugin: 'eclipse'
 apply plugin: 'groovy'
 apply plugin: 'maven'
-apply plugin: 'git'
```
Comment (zhangkun83) [inline build.gradle]: Not sure about git, but you should probably keep idea and eclipse for IDE users.

Comment (jaredsburrows) [inline build.gradle]: @zhangkun83 You arent using anything from these plugins.

File: testProject/build.gradle (hunk 1 above)
Comment (jaredsburrows) [inline testProject/build.gradle]: I'll revert.

File: testProjectAndroid/build.gradle [hunk 3]
```diff
@@ -113,70 +105,70 @@ afterEvaluate {
   }
 
   test.doLast {
-    org.junit.Assert.assertEquals(
-        ['generateArmFreeappDebugAndroidTestProto', 'generateArmFreeappDebugProto',
+    assert ['generateArmFreeappDebugAndroidTestProto', 'generateArmFreeappDebugProto',
```
Comment (jaredsburrows) [inline testProjectAndroid/build.gradle]: Sure, I can revert.

File: build.gradle (hunk 2 above)
Comment (zhangkun83) [inline build.gradle]: I am not using any IDE, but according to the [documentation](https://docs.gradle.org/current/userguide/idea_plugin.html) the plugins generate tasks that will generate IDE-specific configuration files, so I assume they are useful for somebody. At least they were added (which was prior to me working on this plugin) for a reason.

Comment (jaredsburrows) [inline build.gradle]: @zhangkun83 Yeah that's right. I'll add this back as well.

**Comment by jaredsburrows** [comment]
@zhangkun83 I have made the fixes. Let me know if you want any other changes.

**Comment by jaredsburrows** [comment]
@zhangkun83 Was there anything else that I needed to update?

**Comment by zhangkun83** [comment]
@jaredsburrows I need to find some time to take a closer look. Please bear with me.

**Comment by jaredsburrows** [comment]
@zhangkun83 Ok. Take your time!

**Comment by zhangkun83** [comment]
LGTM

**Comment by jaredsburrows** [comment]
Great!

**Comment by jaredsburrows** [comment]
@zhangkun83 Ready to merge?

**Comment by ejona86** [comment]
LGTM
